pkg/controller/controller_ref_manager.go

```diff
@@ -183,6 +183,10 @@ func (m *PodControllerRefManager) ClaimPods(pods []*v1.Pod, filters ...func(*v1.
 
 	match := func(obj metav1.Object) bool {
 		pod := obj.(*v1.Pod)
+		// Skip pods marked for deletion
```

Comment:
That's a good point. `IsPodActive()` checks for both `DeletionTimestamp` and certain `Phase` values. I don't recall whether the breakage of DaemonSet (and I think one other) was because the controllers expected all Phases to be returned in the list, or because they expected deletion-pending Pods to be returned.

I just preserved the existing behavior at the time: some controllers were filtering by `IsPodActive()` and others weren't. If none of the controllers actually care about deletion-pending Pods being returned, then I agree `ClaimPods()` is a reasonable place to do this filtering (but outside `match()` as discussed above).